Examples/Swift/ViewController.swift

```diff
@@ -148,6 +148,12 @@ class ViewController: UIViewController, MGLMapViewDelegate, NavigationViewContro
         navigationViewController.voiceController?.volume = 0.5
         navigationViewController.navigationDelegate = self
         
+        // Uncomment to apply custom styles
+//        styleForRegular().apply()
+//        styleForCompact().apply()
+//        styleForiPad().apply()
+//        styleForCarPlay().apply()
```

Comment:
> What about moving these properties to NavigationViewController as they are, though? 

I guess it makes sense for sizes and dimensions but one drawback is the life cycle of the NavigationViewController. It means that you have to:
1. Initialize the NavigationViewController
2. Present it
3. Now you can modify the constraints, the outlets are hooked up (except for the ones in the page view and other dynamic views that are loaded after)

When using the appearance proxy, you can apply the style whenever, even in the app delegate.

> Is it necessary to separate them into a separate Style class?

A unique class or a unique container view is needed.

> removing NavigationUI moves us slightly away from the idea of separating presentation from functionality.

NavigationUI was kinda separated but the singleton manager approach is rarely recommended. The appearance however, is completely de-coupled. The view doesn't even use a reference to the style, except for custom objects like our style layers.